### ISSUE TITLE
Feature request: README specifically for the examples/ folder

### ISSUE THREAD

**Issue by paul-hammant** [vuejs/test-utils#2030]
**Is your feature request related to a problem? Please describe.**

A way of running the specs in examples/ isn't detailed in the README.

**Describe the solution you'd like**

Add a new `README.md` inside `examples/` that talks about `TodoApp.spec.js` and 
`TodoApp.vue` from the point of view of serving and running the tests.  Sure there's a tutorial online, but the repo would be even better if it had advice on how to run the specs.  Running `yarn test` from the root folder does not include the todo app specs, yet it does run (and all those tests pass).

**Describe alternatives you've considered**

I tried cd-ing to the examples folder and running `yarn test` there but it collects tests for execution as if it were in the root folder.

**Comment by cexbrayat** [comment]
Hi @paul-hammant 

The way to run the example is explained at the top of the test, see https://github.com/vuejs/test-utils/blob/main/examples/TodoApp.spec.js

Adding a README could be nice: do you want to make a PR? Or if the explanation at the top of the test is enough, then we can close this.

**Comment by paul-hammant** [comment]
I'd missed that of course .. so used to just seeing copyright/left there.

I'm readying a pull request for y'all, but I'm seeing something else picked up by that command:

```
 ✓ tests/docs-examples/teleport.spec.ts (1)

 Test Files  1 passed (1)
      Tests  1 passed (1)
   Start at  08:22:59
   Duration  1.24s (transform 307ms, setup 21ms, collect 375ms, tests 31ms, environment 454ms, prepare 99ms)


 PASS  Waiting for file changes...
       press h to show help, press q to quit
✨  Done in 29.19s.
```

**Comment by lmiller1990** [comment]
This was one of the first files I added to this repo - I had totally forgotten about it. We can definitely include more "real" examples, if that's something people would find useful.

There's some other repos with examples, if you'd like to see more, for example:

- https://github.com/lmiller1990/vue-testing-handbook/tree/master/demo-app-vue-3

**Comment by paul-hammant** [comment]
Taking a look, @lmiller1990, but there's errors while running tests after `yarn install`

```
demo-app-vue-3 paul$ yarn test:unit 
yarn run v1.22.19
$ vue-cli-service test:unit
 PASS  tests/unit/ComponentWithVuex.spec.js
 PASS  tests/unit/ComponentWithButtons.spec.js
 PASS  tests/unit/FormSubmitter.spec.js

  ●  Cannot log after tests are done. Did you forget to wait for something async in your test?
    Attempted to log "[Vue Router warn]: Unexpected error when starting the router: TypeError: Cannot read properties of null (reading '_history')
        at Window.get history [as history] (/vcs/vue-testing-handbook/demo-app-vue-3/node_modules/jsdom/lib/jsdom/browser/Window.js:207:79)
        at finalizeNavigation (/vcs/vue-testing-handbook/demo-app-vue-3/node_modules/vue-router/dist/vue-router.cjs.js:2439:41)
        at /vcs/vue-testing-handbook/demo-app-vue-3/node_modules/vue-router/dist/vue-router.cjs.js:2326:27
        at processTicksAndRejections (node:internal/process/task_queues:96:5)".

      at Window.get history [as history] (node_modules/jsdom/lib/jsdom/browser/Window.js:207:79)
      at finalizeNavigation (node_modules/vue-router/dist/vue-router.cjs.js:2439:41)
      at node_modules/vue-router/dist/vue-router.cjs.js:2326:27
      at processTicksAndRejections (node:internal/process/task_queues:96:5)".
      at console.warn (node_modules/@jest/console/build/BufferedConsole.js:224:10)
      at warn (node_modules/vue-router/dist/vue-router.cjs.js:46:18)
      at node_modules/vue-router/dist/vue-router.cjs.js:2618:21
      at processTicksAndRejections (node:internal/process/task_queues:96:5)


  ●  Cannot log after tests are done. Did you forget to wait for something async in your test?
    Attempted to log "[Vue Router warn]: Unexpected error when starting the router: TypeError: Cannot read properties of null (reading '_history')
        at Window.get history [as history] (/vcs/vue-testing-handbook/demo-app-vue-3/node_modules/jsdom/lib/jsdom/browser/Window.js:207:79)
        at finalizeNavigation (/vcs/vue-testing-handbook/demo-app-vue-3/node_modules/vue-router/dist/vue-router.cjs.js:2439:41)
        at /vcs/vue-testing-handbook/demo-app-vue-3/node_modules/vue-router/dist/vue-router.cjs.js:2326:27
        at processTicksAndRejections (node:internal/process/task_queues:96:5)".

      at Window.get history [as history] (node_modules/jsdom/lib/jsdom/browser/Window.js:207:79)
      at finalizeNavigation (node_modules/vue-router/dist/vue-router.cjs.js:2439:41)
      at node_modules/vue-router/dist/vue-router.cjs.js:2326:27
      at processTicksAndRejections (node:internal/process/task_queues:96:5)".
      at console.warn (node_modules/@jest/console/build/BufferedConsole.js:224:10)
      at warn (node_modules/vue-router/dist/vue-router.cjs.js:46:18)
      at node_modules/vue-router/dist/vue-router.cjs.js:2618:21
      at processTicksAndRejections (node:internal/process/task_queues:96:5)

 PASS  tests/unit/Posts.spec.js
  ● Console

    console.warn node_modules/vue-router/dist/vue-router.cjs.js:46
      [Vue Router warn]: No match found for location with path ""
    console.warn node_modules/vue-router/dist/vue-router.cjs.js:46
      [Vue Router warn]: No match found for location with path ""
    console.warn node_modules/vue-router/dist/vue-router.cjs.js:46
      [Vue Router warn]: No match found for location with path "/posts/1"
    console.warn node_modules/vue-router/dist/vue-router.cjs.js:46
      [Vue Router warn]: No match found for location with path ""
    console.warn node_modules/vue-router/dist/vue-router.cjs.js:46
      [Vue Router warn]: No match found for location with path "/posts/new"
    console.warn node_modules/vue-router/dist/vue-router.cjs.js:46
      [Vue Router warn]: No match found for location with path ""
    console.warn node_modules/vue-router/dist/vue-router.cjs.js:46
      [Vue Router warn]: No match found for location with path "/posts/1"

 PASS  tests/unit/router-hooks.spec.js
 PASS  tests/unit/NestedRoute.spec.js
 PASS  tests/unit/SubmitButton.spec.js
 FAIL  tests/unit/Parent.spec.js
  ● Parent › does not render a span

    TypeError: wrapper.find(...).isVisible is not a function

       9 |     const wrapper = mount(Parent)
      10 | 
    > 11 |     expect(wrapper.find("span").isVisible()).toBe(false)
         |                                 ^
      12 |   })
      13 | 
      14 |   it("does render a span", () => {

      at Object.<anonymous> (tests/unit/Parent.spec.js:11:33)
      at processTicksAndRejections (node:internal/process/task_queues:96:5)

  ● Parent › does render a span

    TypeError: wrapper.find(...).isVisible is not a function

      19 |     })
      20 | 
    > 21 |     expect(wrapper.find("span").isVisible()).toBe(true)
         |                                 ^
      22 |   })
      23 | 
      24 |   it("does not render a Child component", () => {

      at Object.<anonymous> (tests/unit/Parent.spec.js:21:33)
      at processTicksAndRejections (node:internal/process/task_queues:96:5)

 PASS  tests/unit/CompositionApi.spec.js
 PASS  tests/unit/Bilingual.spec.js
 PASS  tests/unit/Emitter.spec.js
 PASS  tests/unit/example.spec.js
 PASS  tests/unit/mutations.spec.js
 PASS  tests/unit/App.spec.js
 PASS  tests/unit/getters.spec.js
 PASS  tests/unit/Greeting.spec.js
 PASS  tests/unit/NumberRenderer.spec.js
 FAIL  tests/unit/ParentWithAPICallChild.spec.js
  ● Test suite failed to run

    Cannot find module 'axios' from 'ComponentWithAsyncCall.vue'

      1 | <template>
    > 2 |   <div></div>
        | ^
      3 | </template>
      4 | 
      5 | <script>

      at Resolver.resolveModule (node_modules/jest-resolve/build/index.js:259:17)
      at Object.<anonymous> (src/components/ComponentWithAsyncCall.vue:2:1)

 FAIL  tests/unit/actions.spec.js
  ● Test suite failed to run

    Cannot find module 'axios' from 'actions.spec.js'

       5 | let mockError = false
       6 | 
    >  7 | jest.mock("axios", () => ({
         |      ^
       8 |   post: (_url, _body) => { 
       9 |     return new Promise((resolve) => {
      10 |       if (mockError) 

      at Resolver.resolveModule (node_modules/jest-resolve/build/index.js:259:17)
      at Object.<anonymous> (tests/unit/actions.spec.js:7:6)

Test Suites: 3 failed, 16 passed, 19 total
Tests:       2 failed, 35 passed, 37 total
Snapshots:   0 total
Time:        2.842s, estimated 4s
Ran all test suites.
error Command failed with exit code 1.
```

**Comment by paul-hammant** [comment]
As it happens I found my way with someone else's example - https://github.com/paul-hammant/testing-vue-3-apps-with-vue-test-utils. I went on to add a playwright.js tests, too.  My next problem is I need to see more examples of mock $router setup for unit tests. That's outside the remit of #2030 though, and it could be that one of many many StackOverflow questions/answers on the topic drives me to the right example code.
